ldms/src/core/ldms.c

```diff
@@ -366,6 +370,8 @@ size_t __ldms_format_set_meta_as_json(struct ldms_set *set,
 	size_t cnt;
 	char dbuf[2*LDMS_DIGEST_LENGTH+1];
 	ldms_digest_t digest = ldms_set_digest_get(set);
+	state_buf ss = alloca(sizeof *ss);
+	perm_buf ps = alloca(sizeof *ps);
```

Comment:
This should just be char ss[4], and char ps[16] or whatever you want. But all of this extra syntax is just confusing for other people wondering what all the fuss is about.